samza-core/src/main/scala/org/apache/samza/job/local/ThreadJobFactory.scala

```diff
@@ -73,7 +73,10 @@ class ThreadJobFactory extends StreamJobFactory with Logging {
     ChangelogStreamManager.createChangelogStreams(jobModel.getConfig, jobModel.maxChangeLogStreamPartitions)
 
     val containerId = "0"
-    val jmxServer = new JmxServer
+    var jmxServer: JmxServer = null
+    if (new ClusterManagerConfig(config).getJmxEnabled) {
+      jmxServer = new JmxServer();
```

Comment:
Since we're supporting turn off of JMX  for security reasons, it will be beneficial to add a trivial unit-test here, just to safe-guard against cases where this if-guard is unintentionally removed.

Reply:
If this was passed a method param, I can use mockito for it. But i don't feel its worth to do that refactor to add a unit test for this. 

"Just to safe-guard against cases where this if-guard is unintentionally removed in the future" 
code reviews ?